### ISSUE TITLE
RFE: please include generated man pages into dist tar ball

### ISSUE THREAD

**Issue by kloczek** [dankamongmen/notcurses#981]
Most of the source dist tar balls have included in source dist form roff files.
Wasteing time to convert md to roff only to have installed manpages is a bit pointless.

**Comment by kloczek** [comment]
Other thing is that xml/docbook is much more versatile source format than md.

**Comment by dankamongmen** [comment]
This seems reasonable on its face, but there are a few annoying factors:

(1) the primary source tarball is automatically generated by github
(2) the +dfsg one is made by debian's uscan() and is not allowed to have generated files in it

so I'd need make a third tarball, unless i can somehow modify the first one via github actions or something. i'll keep this open and think about it.

**Comment by dankamongmen** [comment]
BTW, if you're just trying to avoid *building* the man pages, you can run `cmake -DUSE_PANDOC=off`. Of course, then you won't have them at all.

**Comment by kloczek** [comment]
Yes but would produce resources without man pages :P

Really I think hat it would be do one tome conversion to xm/docbook.

FYI I'm trying to minimise whole OS distro tooling. pan doc is written in haskel which drags with notcurses man pages huge constellation of other packages. xsltproc is written ion C. Whole gnome uses xml and this allows easy process xml tree into any possible form. MD provides only easy to edit form.
I think that it would good to maybe switch to some other md->roff converter but I don't see good candidate. From other side there are now few editors which provides visual form of xml/docbook editing layer.

In other words above should provide you some right context why I've decided to bother you about this subject :)

**Comment by kloczek** [comment]
I want only to say that if you see some other way to reach those (currently) contradicting targets just let me know.
For now nothing needs to be solved. We can just discuss only ..

**Comment by dankamongmen** [comment]
@kloczek how would this work for you: the released tarballs would not change (i.e. they would only have Pandoc source), but there would be a new tarball containing the generated man pages. if you wanted to build notcurses without *building* the man pages, but wanted to *install* the man pages, you could then accomplish it via:

* download notcurses tarball
* download generated man pages tarball
* build with `USE_PANDOC=off`
* install results
* install generated man pages

would that accomplish your goals? due to Debian requirements, this is easiest for me. i'd then make creation of the generated man page tarball part of the automated [notcurses release process](https://github.com/dankamongmen/notcurses/blob/master/tools/release.sh).

**Comment by dankamongmen** [comment]
OK, I'm going to do that. The doc package will be uploaded as a github asset, just like the dfsg tarball and the signatures. If you want to download it rather than building the man/html pages (and thus requiring pandoc), it'll be there.

**Comment by ryandesign** [comment]
> (1) the primary source tarball is automatically generated by github

Tarballs generated on-demand by `git archive` are not great to rely on since [GitHub does not guarantee their immutability](https://github.blog/open-source/git/update-on-the-future-stability-of-source-code-archives-and-hashes/). Uploading your manually-created source code archive (including generated files) to each release is recommended.

**Comment by dankamongmen** [comment]
> > (1) the primary source tarball is automatically generated by github
> 
> Tarballs generated on-demand by `git archive` are not great to rely on since [GitHub does not guarantee their immutability](https://github.blog/open-source/git/update-on-the-future-stability-of-source-code-archives-and-hashes/). Uploading your manually-created source code archive (including generated files) to each release is recommended.

hrmmm. valid. let's get a new bug on this.
